backend/local/backend.go

```diff
@@ -97,6 +119,150 @@ func (b *Local) Configure(c *terraform.ResourceConfig) error {
 	return f(c)
 }
 
+func (b *Local) States() ([]string, string, error) {
+	// If we have a backend handling state, defer to that.
+	if b.Backend != nil {
+		if b, ok := b.Backend.(backend.MultiState); ok {
+			return b.States()
+		} else {
+			return nil, "", ErrEnvNotSupported
+		}
+	}
+
+	// the listing always start with "default"
+	envs := []string{backend.DefaultStateName}
+
+	current, err := b.currentStateName()
+	if err != nil {
+		return nil, "", err
+	}
+
+	entries, err := ioutil.ReadDir(filepath.Join(b.workingDir, DefaultEnvDir))
+	// no error if there's no envs configured
+	if os.IsNotExist(err) {
+		return envs, backend.DefaultStateName, nil
+	}
+	if err != nil {
+		return nil, "", err
+	}
+
+	currentExists := false
+	var listed []string
+	for _, entry := range entries {
+		if entry.IsDir() {
+			name := filepath.Base(entry.Name())
+			if name == current {
+				currentExists = true
+			}
+			listed = append(listed, name)
+		}
+	}
+
+	// current was out of sync for some reason, so return defualt
+	if !currentExists {
+		current = backend.DefaultStateName
+	}
+
+	sort.Strings(listed)
+	envs = append(envs, listed...)
+
+	return envs, current, nil
+}
+
+// DeleteState removes a named state.
+// The "default" state cannot be removed.
+func (b *Local) DeleteState(name string) error {
+	// If we have a backend handling state, defer to that.
+	if b.Backend != nil {
+		if b, ok := b.Backend.(backend.MultiState); ok {
+			return b.DeleteState(name)
+		} else {
+			return ErrEnvNotSupported
+		}
+	}
+
+	if name == "" {
+		return errors.New("empty state name")
+	}
+
+	if name == backend.DefaultStateName {
+		return errors.New("cannot delete default state")
+	}
+
+	_, current, err := b.States()
+	if err != nil {
+		return err
+	}
+
+	// if we're deleting the current state, we change back to the default
+	if name == current {
```

Comment:
Oh yeas, I missed that and figured it OK since state was protected if it's not empty. Easy change.